cli/internal/lockfile/berry_lockfile.go

```diff
@@ -493,8 +493,13 @@ func (d *_Descriptor) primaryVersion() (string, bool) {
 	if patchFileIndex < 0 || versionRangeIndex < 0 {
 		panic("Patch reference is missing required markers")
 	}
+	// The ':' following npm protocol gets encoded as '%3A' in the patch string
+	version := strings.Replace(d.versionRange[versionRangeIndex+1:patchFileIndex], "%3A", ":", 1)
```

Comment:
I ran into this issue when trying to get this working, but it also came up in #2146